pkg/server/config.go

```diff
@@ -263,6 +264,14 @@ func NewConfig(opts *kcpserveroptions.CompletedOptions) (*Config, error) {
 		return nil, err
 	}
 
+	var shardVirtualWorkspaceURL *url.URL
+	if !opts.Virtual.Enabled && opts.Extra.ShardVirtualWorkspaceURL != "" {
```

Comment:
should we return an error if the virtual is enabled and the URL was provided?

Reply:
Probably. Not sure it belongs to this PR though. wdyt ?

Reply:
Sure, can be done in a separate PR, just saying that this might be a configuration error.